src/gui/PianoView.cpp

```diff
@@ -618,6 +619,9 @@ void PianoView::mouseMoveEvent( QMouseEvent * _me )
  */
 void PianoView::keyPressEvent( QKeyEvent * _ke )
 {
+#ifdef LMMS_BUILD_LINUX
```

Comment:
I only need the codes for Windows and Mac. Generally, Qt has special Qt-names for keys (for example, Qt::Key_Shift ) for use with QKeyEvent::key() that do not depend on the platform. And it can be used to determine the key. However, in this case it is impossible to determine which shift key is pressed, left or right.